src/analyze/analyze-security.c

```diff
@@ -613,6 +616,43 @@ static int assess_ip_address_allow(
         return 0;
 }
 
+static int assess_ip_filters_custom(
+                const struct security_assessor *a,
+                const struct security_info *info,
+                const void *data,
+                uint64_t *ret_badness,
+                char **ret_description) {
+
+        char *d = NULL;
+        uint64_t b;
+
+        assert(info);
+        assert(ret_badness);
+        assert(ret_description);
+
+        if (!info->ip_filters_custom_ingress && !info->ip_filters_custom_egress) {
+                d = strdup("Service does not define custom IP filters with BPF programs");
+                b = 1;
+        } else if (info->ip_filters_custom_ingress && info->ip_filters_custom_egress) {
+                d = strdup("Service defines custom IP filters with BPF programs");
+                b = 0;
+        } else if (info->ip_filters_custom_ingress) {
+                d = strdup("Service defines a custom ingress IP filter with a BPF program");
+                b = 0;
+        } else {
+                d = strdup("Service defines a custom egress IP filter with a BPF program");
+                b = 0;
+        }
```

Comment:
hmm, this should be tweaked I think. If a service has IPAddressAllow= defined, then not having IPFilterIngress= shouldn't be considered "bad". Similar, if IPAddressAllow= not being set shouldn't count as bad if IPFilterIngress= is set. i.e. the two tests should check for each other (but should be kept as two entries in the output). This requires a bit of care though, as if neither is set, we don't want both to account for the badness...